### PR TITLE
Update AnalysisService ARM schema and test cases

### PR DESCRIPTION
**Pull request by YingZhang-MSPBI**
Update AnalysisService ARM schema according latest version and also updated the test cases for the schema.

### CODE REVIEW

**Comment by minghan** [comment]
I am from the Azure Analysis Services team and this change looks good for check in. Thank you. 👍

File: schemas/2017-08-01/Microsoft.AnalysisServices.json [hunk 1]
```diff
@@ -0,0 +1,219 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2017-08-01/Microsoft.AnalysisServices.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.AnalysisServices",
+  "description": "SQL Analysis Services on Azure",
+  "resourceDefinitions": {
+    "servers": {
```
Comment (TianoMS) [inline schemas/2017-08-01/Microsoft.AnalysisServices.json]: >servers [](start = 5, length = 7)

please add the new resource types to 2015-01-01/deploymentTemplate and 2014-04-01-preview/deploymentTemplate.

Comment (YingZhang-MSPBI) [reply schemas/2017-08-01/Microsoft.AnalysisServices.json]: Already add into templates. Thank you.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 2]
```diff
@@ -698,6 +698,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-05-01-preview/Microsoft.Insights.json#/resourceDefinitions/ProactiveDetectionConfigs" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-07-01/Microsoft.ContainerService.json#/resourceDefinitions/containerServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-03-31/Microsoft.ContainerService.json#/resourceDefinitions/managedClusters" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-08-01/Microsoft.AnalysisServices.json#/resourceDefinitions/servers" },
```
Comment (minghan) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: this line should be added after line 48.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 3]
```diff
@@ -710,6 +710,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-05-01-preview/Microsoft.Insights.json#/resourceDefinitions/ProactiveDetectionConfigs" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-07-01/Microsoft.ContainerService.json#/resourceDefinitions/containerServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-03-31/Microsoft.ContainerService.json#/resourceDefinitions/managedClusters" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-08-01/Microsoft.AnalysisServices.json#/resourceDefinitions/servers" },
```
Comment (minghan) [inline schemas/2015-01-01/deploymentTemplate.json]: should be added after line 59.

File: schemas/2017-08-01/Microsoft.AnalysisServices.json (hunk 1 above)
Comment (minghan) [reply schemas/2017-08-01/Microsoft.AnalysisServices.json]: @YingZhang-MSPBI : I don't see the change.

**Comment by minghan** [comment]
@YingZhang-MSPBI : please add a description to this pull request. Thanks.

**Comment by YingZhang-MSPBI** [comment]
@minghan 
I have moved lines to correct place and add the description. Thank you!

**Comment by minghan** [comment]
@TianoMS : can this PR be approved? Thank you!

File: schemas/2017-08-01/Microsoft.AnalysisServices.json [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+          "type": "string",
+          "enum": [
+            "Microsoft.AnalysisServices/servers"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2016-05-16",
+            "2017-07-14",
```
Comment (TianoMS) [inline schemas/2017-08-01/Microsoft.AnalysisServices.json]: why are multiple versions here?

File: schemas/2017-08-01/Microsoft.AnalysisServices.json [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.AnalysisServices/servers"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2016-05-16",
```
Comment (TianoMS) [inline schemas/2017-08-01/Microsoft.AnalysisServices.json]: >"2016-05-16", [](start = 12, length = 13)

This version is already defined in 2016-05-16/Microsoft.AnalysisServices.json.

File: schemas/2017-08-01/Microsoft.AnalysisServices.json (hunk 4 above)
Comment (YingZhang-MSPBI) [reply schemas/2017-08-01/Microsoft.AnalysisServices.json]: These versions are all supported.

**Pull request merged**